cogito/gchatsink_private_test.go

```diff
@@ -55,7 +58,118 @@ func TestShouldSendToChatCustomConfig(t *testing.T) {
 	}
 }
 
-func TestGChatFormatText(t *testing.T) {
+func TestPrepareChatMessageSuccess(t *testing.T) {
+	type testCase struct {
+		name        string
+		request     PutRequest
+		inputDir    fstest.MapFS
+		wantPresent []string
+		wantAbsent  []string
+	}
+
+	baseRequest := PutRequest{
+		Source: Source{Owner: "the-owner"},
+		Params: PutParams{State: StateError},
+		Env:    Environment{BuildJobName: "the-job"},
+	}
+
+	baseGitRef := "deadbeef"
+
+	basePresent := []string{
+		baseRequest.Source.Owner, baseRequest.Env.BuildJobName, baseGitRef}
+
+	test := func(t *testing.T, tc testCase) {
+		have, err := prepareChatMessage(tc.inputDir, tc.request, baseGitRef)
+
+		assert.NilError(t, err)
+		for _, elem := range tc.wantPresent {
+			assert.Check(t, strings.Contains(have, elem), "wanted: %s", elem)
+		}
+		for _, elem := range tc.wantAbsent {
+			assert.Check(t, !strings.Contains(have, elem), "not wanted: %s", elem)
+		}
+	}
+
+	testCases := []testCase{
+		{
+			name:        "default build summary",
+			request:     baseRequest,
+			wantPresent: basePresent,
+		},
+		{
+			name: "chat_message overrides default",
+			request: testhelp.MergeStructs(
+				baseRequest,
+				PutRequest{Params: PutParams{ChatMessage: "the-custom-message"}}),
+			wantPresent: []string{"the-custom-message"},
+			wantAbsent:  basePresent,
+		},
+		{
+			name: "chat_message and append",
+			request: testhelp.MergeStructs(
+				baseRequest,
+				PutRequest{
+					Params: PutParams{
+						ChatMessage:       "the-custom-message",
+						ChatMessageAppend: true,
+					},
+				}),
+			wantPresent: append(basePresent, "the-custom-message"),
+		},
+		{
+			name: "chat message file",
+			request: testhelp.MergeStructs(
+				baseRequest,
+				PutRequest{Params: PutParams{ChatMessageFile: "registration/msg.txt"}}),
+			inputDir: fstest.MapFS{
+				"registration/msg.txt": {Data: []byte("from-custom-file")},
```

Comment:
It creates a memory-based file system, because `fstest.MapFS` implements the `io/fs` interface.
If you follow the path to the type of `inputDir` in this test and then to the type of the parameter in the SUT (system under test), `prepareChatMessage()` it should be clearer. In this PR there is also a comment for `prepareChatMessage()`.